### PR TITLE
1689 Add subphases to lb_iter example

### LINKED ISSUE #1689: Add subphases to lb_iter example
**What Needs to be Done?**
There are no vt examples that use subphases. Add subphases to the `lb_iter` example.

### PR DESCRIPTION
**Pull request by nlslatt**
Fixes #1689

### CODE REVIEW

File: examples/collection/lb_iter.cc [hunk 1]
```diff
@@ -51,12 +51,15 @@ struct IterCol : vt::Collection<IterCol, vt::Index1D> {
 
   struct IterMsg : vt::CollectionMessage<IterCol> {
     IterMsg() = default;
-    explicit IterMsg(int64_t const in_work_amt, int64_t const in_iter)
-      : iter_(in_iter), work_amt_(in_work_amt)
+    explicit IterMsg(
```
Comment (PhilMiller) [inline examples/collection/lb_iter.cc:54]: This probably doesn't need to be marked `explicit`, since it takes multiple arguments

Comment (PhilMiller) [reply examples/collection/lb_iter.cc:54]:
```suggestion
    IterMsg(
```

**Review by PhilMiller — APPROVED** [review]

File: examples/collection/lb_iter.cc (hunk 1 above)
Comment (nlslatt) [reply examples/collection/lb_iter.cc]: Changed.

**Comment by jstrzebonski** [comment]
I've just fixed `Check PR description`.

**Review by lifflander — APPROVED** [review]

**Pull request merged**